routing/bandwidth.go

```diff
@@ -143,6 +143,14 @@ func (b *bandwidthManager) getBandwidth(cid lnwire.ShortChannelID,
 					"auxiliary bandwidth: %w", err))
 			}
 
+			// If AuxBandwidth returns None, it means the external
+			// traffic shaper is not handling the channel. In that
+			// case, we'll just return the original bandwidth and
+			// no custom amount.
+			if auxBandwidth.IsNone() {
```

Comment:
I suggest removing the HTLC amount from the bandwidthResult, because if its not a custom payment we just check for the amount which was passed in I think that's what the channel needs to route at a bare minimum anyways.

Reply:
No, unfortunately we can't do that. If we're trying to pay an invoice over let's say 500k sats, then that amount is passed in as `amount`. In a channel that only has the capacity of 100k sats, we clearly can't send 500k. And with an asset channel, we're not going to, as we're going to convert to assets and will only really send 354 sats. That's why the bandwidth result overwrites the `htlcAmount` with 0 (which just means "the minimum amount possible") in case of an asset channel.

What we need to fix here is the behavior for non-asset channels, which is what this commit/PR does.

Reply:
ahh gottya, so when sending a custom payment the HTLC amount here not already adjusted for, ok understand.